Utilities/Thread.cpp

```diff
@@ -2774,24 +2803,10 @@ u64 thread_ctrl::get_affinity_mask(thread_class group)
 		}
 		case native_core_arrangement::intel_ht:
 		{
-			/* This has been disabled as it seems to degrade performance instead of improving it.
-			if (thread_count <= 4)
-			{
-				//i3 or worse
-				switch (group)
-				{
-				case thread_class::rsx:
-				case thread_class::ppu:
-					return (0b0101 & all_cores_mask);
-				case thread_class::spu:
-					return (0b1010 & all_cores_mask);
-				case thread_class::general:
-					return all_cores_mask;
-				}
-			}
-			*/
-
-			return all_cores_mask;
+			if (g_cfg.core.thread_scheduler == thread_scheduler_mode::ht_off || g_cfg.core.thread_scheduler == thread_scheduler_mode::old && thread_count >= 12)
```

Comment:
```suggestion
			if (g_cfg.core.thread_scheduler == thread_scheduler_mode::ht_off || (g_cfg.core.thread_scheduler == thread_scheduler_mode::old && thread_count >= 12))
```

Reply:
Wouldn't that make the thread count check only apply to the old scheduler?

Reply:
this should be the same boolean logic, just with () for readability and to prevent warnings

Reply:
I guess you meant to write this:
if (thread_count >= 12  && (g_cfg.core.thread_scheduler == thread_scheduler_mode::ht_off || g_cfg.core.thread_scheduler == thread_scheduler_mode::old))

Reply:
Yes.

Reply:
remember, && has precedence over ||